spec.cue

```diff
@@ -0,0 +1,61 @@
+package spec
+
+version: 2
+sources: [...#Source]
+
+
+#StringInt: string | number
+
+#Partition: {
+	name:        string & !=""
+	data_type:   string & !=""
+	expression?: string
+	vals?: {
+		macro: string & !=""
+		args: [string]: #StringInt
+	}
+	path_macro?: string & !=""
+}
+
+#SourceSnowflake: {
```

Comment:
This version creates a separate `Source` for each of the supported  sources:

- Snowflake
- Redshift
- Spark

Reply:
We've since added support for BigQuery, Synapse, and Azure SQL... it's been a busy couple days in this package